ivy/functional/backends/jax/general.py

```diff
@@ -208,8 +208,8 @@ def inplace_update(
     keep_input_dtype: bool = False,
 ) -> ivy.Array:
     if ivy.is_array(x) and ivy.is_array(val):
-        if ensure_in_backend:
-            raise ivy.utils.exceptions.IvyException(
+        if ensure_in_backend or ivy.is_native_array(x):
```

Comment:
Lets add the check like `if ensure_in_backend or (ivy.is_native_array(x) and not ivy.get_array_mode())`

I guess it would be better to raise an error only if tensor was passed when array_mode was False, otherwise, we will be returning an ivy array anyway so I guess its fine in that case not to raise an error.
and same case with tensorflow